internal/collections/map_test.go

```diff
@@ -106,3 +106,26 @@ func TestNewCaseSensitiveKeyMap(t *testing.T) {
 	}
 
 }
+
+func BenchmarkTxSetGet(b *testing.B) {
+	c := NewCaseSensitiveKeyMap(variables.RequestHeaders)
+	b.Run("Set", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			key := fmt.Sprintf("key%d", i)
```

Comment:
I think the interpolation here impacts the benchmark. I would generate the keys in a map first.

Reply:
Fixed it 👍